packages/block-library/src/gallery/edit.js

```diff
@@ -541,7 +544,7 @@ export default function GalleryEdit( props ) {
 	};
 
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
-		defaultBlock: DEFAULT_BLOCK,
+		allowedBlocks: [ 'core/image', 'core/video' ],
 		directInsert: true,
```

Comment:
Let's define `allowedBlocks` as a constant outside the component, since it's a static value - `const ALLOWED_BLOCKS = [ 'core/image', 'core/video' ]`.

The `directInsert` option can be removed now. It's only useful when inner blocks only allow a single block type.